File: iOS/Baccarat Parveen/Baccarat/Baccarat/Deck.m
//
//  Deck.m
//  Baccarat
//
//  Created by Parveen Khatkar on 8/2/15.
//  Copyright (c) 2015 Codetrix Studio. All rights reserved.
//

#import "Deck.h"
#import "Card.h"

@implementation Deck

-(instancetype)init
{
    self.cards = [[NSMutableArray alloc] init];
    const char *suits = [@"scdh" UTF8String];
    
    for (int j = 0; j <4; j++)
    {
        for (int i = 1; i <= 13; i++)
        {
            [self.cards addObject:[[Card alloc] initWithSuit:suits[j] Card:i]];
        }
    }
    return self;
}

@end

File: Windows Phone/StrategySimulator/StrategySimulator/Model/Deck.cs

namespace StrategySimulator.Model
{
    public class Deck
    {
        public Card[] Cards { get; private set; }
        private int _cardsInDeck = 52;

        public Deck()
        {
            createDeck();
        }

        private void createDeck()
        {
            int index = 0;
            Cards = new Card[_cardsInDeck];
            for (int suit = 1; suit < 5; suit++)
            {
                for (int value = 1; value < 14; value++)
                {
                    Card card = new Card((Suits)suit, (Values)value);
                    Cards[index++] = card;
                }
            }
        }
    }
}
